ste/xfer-anyToRemote.go

```diff
@@ -30,12 +31,78 @@ import (
 	"sync"
 
 	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+
 	"github.com/Azure/azure-storage-azcopy/common"
 )
 
 // This sync.Once is present to ensure we output information about a S2S access tier preservation failure to stdout once
 var s2sAccessTierFailureLogStdout sync.Once
 
+// This sync.Once and string pair ensures that we only get a user's destination account kind once when handling set-tier
+// Premium block blob doesn't support tiering, and page blobs only support P1-80.
+// There are also size restrictions on tiering.
+var destAccountSKU string
+var destAccountKind string
+var tierSetPossibleFail bool
+var getDestAccountInfo sync.Once
+
+func prepareDestAccountInfo(bURL azblob.BlobURL, jptm IJobPartTransferMgr, ctx context.Context, mustGet bool) {
+	getDestAccountInfo.Do(func() {
+		infoResp, err := bURL.GetAccountInfo(ctx)
+		if err != nil {
+			// If GetAccountInfo fails, this transfer should fail because we lack at least one available permission
+			// UNLESS the user is using OAuth. In which case, the account owner can still get the info.
+			// If we fail to get the info under OAuth, don't fail the transfer.
+			// (https://docs.microsoft.com/en-us/rest/api/storageservices/get-account-information#authorization)
+			if mustGet {
+				jptm.FailActiveSendWithStatus("Checking destination tier availability (Set PageBlob tier) ", err, common.ETransferStatus.TierAvailabilityCheckFailure())
```

Comment:
It seems odd to use FailActiveSendWithStatus inside a sync.Once.  Would it make more sense just to set up destAccountSKU, destAccountKind and *destAccountTypeError* inside the Once, and then use FailActiveSendWithStatus outside the sync.Once?